ghstack/gpg_sign.py

```diff
@@ -0,0 +1,35 @@
+"""This module determines if the commits need to be signed.
+We need to do this manually, because ghstack uses commit-tree instead of commit.
+commit-tree command doesn't pick up commit.gpgsign git config
+
+The porcelain git behavior w.r.t. signing is
+
+when both `commit.gpgsign` and `user.signingkey` are set, the commit is signed
+when only `commit.gpgsign` is true, git errors out
+
+This module will retain this behavior:
+We will attempt to sign as long as `commit.gpgsign` is true.
+If not key is configure, error will occur
+"""
+from typing import Tuple, Union
+
+import ghstack.shell
+
+_should_sign = None
+
+
+def gpg_args_if_necessary(
+    shell: ghstack.shell.Shell = ghstack.shell.Shell()
```

Comment:
once we have some sort of env plumbing, would feed env into this function and that would make it possible to override this setting in testing